packages/template-retail-react-app/app/pages/checkout/partials/contact-info.jsx

```diff
@@ -62,9 +67,21 @@ const ContactInfo = ({isSocialEnabled = false, isPasswordlessEnabled = false, id
     const [showPasswordField, setShowPasswordField] = useState(false)
     const [signOutConfirmDialogIsOpen, setSignOutConfirmDialogIsOpen] = useState(false)
 
+    const [authModalView, setAuthModalView] = useState(PASSWORD_VIEW)
+    const authModal = useAuthModal(authModalView)
+    const [isPasswordlessLoginClicked, setIsPasswordlessLoginClicked] = useState(false)
+
     const submitForm = async (data) => {
         setError(null)
         try {
+            if (isPasswordlessLoginClicked) {
```

Comment:
I use a `isPasswordlessLoginClicked` state to differentiate when the passwordless login button is clicked. Is there a better way of doing this or is this okay?